modules/n1ql/pages/n1ql-language-reference/metafun.adoc

```diff
@@ -703,6 +703,84 @@ FROM system:completed_requests;
 ----
 ====
 
+[[hashbytes,HASHBYTES()]]
+
+== HASHBYTES(`input`, [ `options` ])
+
+=== Description
+
+This function returns a binary hash value for a given input using a specified hashing algorithm. 
+By using this function, you can protect your data by masking its original form while still allowing verification or comparison. 
+
+=== Arguments
+
+input:: The data to be hashed, represented as a binary object.
++
+If the data is not in the binary form, you must convert it to a JSON marshalled value (JSON string) before it can be hashed. 
+You can use the ENCODE_JSON() function to do this conversion. 
+For more information, see xref:n1ql-language-reference/jsonfun.adoc[JSON Functions].     
```

Comment:
Looking closely at the original ticket, I'm not sure if this is quite right.

```suggestion
input:: A binary object or any {sqlpp} data type.
The JSON marshalled value of the data is used as the input.
```

That's how I read it anyway. If you're in doubt you can test the input by trying out the function with a number, an object, an array, NULL, and MISSING, and see whether you get valid output each time.